### ISSUE TITLE
non-free dependency: "github.com/feyeleanor/sets"

### ISSUE THREAD

**Issue by onlyjob** [docker/goamz#18]
_github.com/feyeleanor/sets_ is [non-free](https://github.com/feyeleanor/sets/issues/1)

This makes _docker-goamz_ non-distributable that in turn affects _docker-distribution_ that recently gained dependency on _docker-goamz_.

**Comment by danielcotton** [comment]
Only used in CloudWatch and completely unmaintained.
+1 for practical reasons.

It looks like [this package](https://github.com/deckarep/golang-set) could be used with some minor changes. Seems to be regular MIT-licensed without the evil clause, that okay with you @onlyjob?

Alternatively, and given the minimal usage, the code could probably be modified to not use sets in the first place.

**Comment by onlyjob** [comment]
Yes, `github.com/deckarep/golang-set` seems to be a good choice: DFSG compatible license and we already have it in [Debian](http://debian.org). ;)

Just please remember to vendor it by tag/release.
